File: download_drift.dart
import 'dart:io';

void main() async {
  final client = HttpClient();
  
  // Download sqlite3.wasm
  try {
    final req1 = await client.getUrl(Uri.parse('https://github.com/simolus3/sqlite3.dart/raw/main/sqlite3/assets/sqlite3.wasm'));
    final res1 = await req1.close();
    await res1.pipe(File('web/sqlite3.wasm').openWrite());
    print('sqlite3.wasm downloaded.');
  } catch (e) {
    print('Failed to download sqlite3.wasm: $e');
  }

  // Download drift_worker.js
  try {
    final req2 = await client.getUrl(Uri.parse('https://raw.githubusercontent.com/simolus3/drift/main/drift/lib/web/worker.dart.js'));
    final res2 = await req2.close();
    await res2.pipe(File('web/drift_worker.js').openWrite());
    print('drift_worker.js downloaded.');
  } catch (e) {
    print('Failed to download drift_worker.js: $e');
  }
}

File: download_drift.py
import urllib.request
import os

print("Downloading sqlite3.wasm...")
url_wasm = "https://github.com/simolus3/sqlite3.dart/releases/download/sqlite3-2.5.0/sqlite3.wasm"
try:
    urllib.request.urlretrieve(url_wasm, "web/sqlite3.wasm")
    print("sqlite3.wasm downloaded.")
except Exception as e:
    print(f"Failed to download sqlite3.wasm: {e}")

print("Downloading drift_worker.js...")
# drift_worker.js is not officially released as a standalone file in recent versions, 
# but we can grab a known working version from an older release or compile it.
# Let's write a dummy worker or use a known URL.
url_worker = "https://raw.githubusercontent.com/simolus3/drift/main/drift/lib/web/worker.dart.js"
try:
    urllib.request.urlretrieve(url_worker, "web/drift_worker.js")
    print("drift_worker.js downloaded.")
except Exception as e:
    print(f"Failed to download drift_worker.js: {e}")
